### DISCUSSION TITLE
Drawer not full screen

### DISCUSSION THREAD

**Discussion by hariria** [chakra-ui/chakra-ui#6533] [General]
Hello!

I wanted to ask if there was a way for the [`<Drawer />` ](https://chakra-ui.com/docs/components/drawer/usage) component to only open within a div rather than the full-screen `<body>`. I have played around with the `<Drawer />` component's `portalProps` and I have set the `containerRef` to not be the document body but I haven't been able to make it so that it doesn't open to the full width  / bottom of the screen.

I'm on the latest Chakra (2.2.8).

Kind of like this:
<img width="813" alt="Screen Shot 2022-08-22 at 9 39 02 AM" src="https://user-images.githubusercontent.com/31301117/185973674-fde24ae5-e2a6-49a8-b43c-513a72612ad7.png">

**Comment by hariria** [comment]
Just wanted to bump this. Does anyone know if it's possible to  do this?

**Comment by lucasrabiec** [comment]
Bumping. I am also wondering if is it possible to open the Drawer on the right side of the div and not on the whole body.

**Comment by hariria** [comment]
@segunadebayo is this possible?

**Comment by TheMechanic** [comment]
Hello ! up ! I'm trying to use Drawer in a specific container, but no way to get something working properly everywhere (safari for example...) It's crazy it doesn't work :/ 
Is there any way to get this, or is it in the roadmap ?

thanks

**Comment by lucasrabiec** [comment]
I figured out a workaround but it works in my case. It needs some props in DrawerOverlay and DrawerContent to be set.

```tsx
 <Drawer>
      <DrawerOverlay
          h='auto'
          top={containerOffsetTop}
          bottom={containerOffsetBottom}
          position='absolute'
      />
      <DrawerContent
        containerProps={{
          h: 'auto',
          top: containerOffsetTop,
          bottom: containerOffsetBottom,
        }}
        style={{ position: 'absolute' }}
      >
        <DrawerBody>{children}</DrawerBody>
      </DrawerContent>
    </Drawer>
```
`containerOffsetTop` and `containerOffsetBottom` are the navbar and footer heights in my specific case.
